test/benchmark/main.cc

```diff
@@ -3,14 +3,32 @@
 #include "test/test_common/environment.h"
 
 #include "benchmark/benchmark.h"
+#include "tclap/CmdLine.h"
 
-// Boilerplate main(), which discovers benchmarks and runs them.
+bool g_skip_expensive_benchmarks = false;
+
+// Boilerplate main(), which discovers benchmarks and runs them. This uses two
+// different flag parsers, so the order of flags matters: flags defined here
+// must be passed first, and flags defined in benchmark::Initialize second,
+// separated by --.
 int main(int argc, char** argv) {
   Envoy::TestEnvironment::initializeTestMain(argv[0]);
 
-  benchmark::Initialize(&argc, argv);
-  if (benchmark::ReportUnrecognizedArguments(argc, argv)) {
-    return 1;
+  TCLAP::CmdLine cmd("envoy-benchmark-test", ' ', "0.1");
```

Comment:
It's annoying that benchmark doesn't use absl flags. Can you add a TODO to drop this if that change ever gets made?

Also note in a comment here that you can't use absl flags here yourself or you'll get errors with the benchmarking flags.

Reply:
Also clang-tidy complains about this line, through no fault of your own.
```
external/com_github_mirror_tclap/include/tclap/Arg.h:482:53: error: Call to virtual method 'Arg::toString' during construction bypasses virtual dispatch [clang-analyzer-optin.cplusplus.VirtualCall,-warnings-as-errors]
                                "Argument flag can only be one character long", toString() ) );
                                                                                ^
test/benchmark/main.cc:17:18: note: Calling constructor for 'CmdLine'
  TCLAP::CmdLine cmd("envoy-benchmark-test", ' ', "0.1");
```
I think (hope) you can resolve this with something like
```
  // NOLINTNEXTLINE(clang-analyzer-optin.cplusplus.VirtualCall)
```

Reply:
thanks, hope that works ;)